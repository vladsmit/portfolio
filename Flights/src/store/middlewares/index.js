import flightMiddleware from './flightMiddleware';

// eslint-disable-next-line
export default [
    flightMiddleware,
];