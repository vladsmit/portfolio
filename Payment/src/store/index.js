import Vue from 'vue';
import Vuex from 'vuex';
import createPersistedState from 'vuex-persistedstate';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    validationPassed: false,
    accounts: []
  },
  
  getters: {
    getValidationStatus(state) {
      return state.validationPassed;
    },

    getAccounts(state) {
      return state.accounts;
    },

    getAccountsLength(state, getters) {
      return getters.getAccounts.length;
    }
  },
  
  mutations: {
    changeValidationPassed(state) {
      state.validationPassed = !state.validationPassed;
    },

    addPayment(state, payload) {
      state.accounts = [...state.accounts, {
        id: state.accounts.length + 1,
        accountNumber: payload.accountNumber,
        sum: payload.sum,
        cardNumber: payload.cardNumber,
        owner: payload.owner,
        date: payload.date
      }];
    }
  },
  
  plugins: [createPersistedState()]
});
