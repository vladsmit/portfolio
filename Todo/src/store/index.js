import Vue from 'vue';
import Vuex from 'vuex';
import Axios from 'axios';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    taskList: [],
    page: null,
  },

  getters: {
    getTaskList(state) {
      return state.taskList;
    },

    getPage(state) {
      return state.page;
    },

    paginationPageList(state) {
      let start = (state.page - 1) * 10;
      let end = state.page * 10;
      return state.taskList.slice(start, end);
    }
  },

  mutations: {
    saveInLocalStorage(state, payload) {
      localStorage.setItem('taskList', JSON.stringify(payload));
    },

    setTaskList(state) {
      if (localStorage.taskList) {
        let taskList = JSON.parse(localStorage.taskList);
        state.taskList = taskList;
      }
    },

    reverseTaskList(state, payload) {
      payload.reverse();
    },

    addTaskInTaskList(state, payload) {
      state.taskList.unshift({
        id: payload.id,
        title: payload.title,
        completed: payload.completed,
        edit: payload.edit
      });
    },

    deleteTaskFromTaskList(state, payload) {
      state.taskList = state.taskList.filter((task) => payload !== task);
    },

    incrementPage(state) {
      state.page++;
    },

    decrementPage(state) {
      state.page--;
    },

    pageAmount(state, payload) {
      state.page = payload.amount;
    }
  },

  actions: {
    async getTasksFromJson({ commit }) {
      let { data } = await Axios.get("../tasks.json");
      commit('reverseTaskList', data);
      commit('saveInLocalStorage', data);
      commit('setTaskList');
    },

    addTaskInLocalStorage({ commit, state }, payload) {
      commit('addTaskInTaskList', payload);
      commit('saveInLocalStorage', state.taskList);
    },

    deleteTaskFromLocalStorage({ commit, state }, payload) {
      commit('deleteTaskFromTaskList', payload);
      commit('saveInLocalStorage', state.taskList);
    },
  }
});