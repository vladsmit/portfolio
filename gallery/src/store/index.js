import Vue from 'vue'
import Vuex from 'vuex'
import Axios from 'axios'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    categories: null,
    loading: false
  },

  getters: {
    getLoadingStatus(state) {
      return state.loading;
    },

    getCategories(state) {
      return state.categories;
    }
  },

  mutations: {
    isLoading(state, payload) {
      state.loading = payload;
    },

    savePhotosInState(state, payload) {
      let photos = [];
      let num = 0;
      for (let i = 0; i < 4; i++) {
        photos.push([]);
        for (let j = num; j < 24; j++) {
          if (photos[i].length < 6) {
            photos[i].push(payload[j]);
          } else {
            break;
          }
        }
        num += 6;
      }
      state.categories = photos;
    },
  },

  actions: {
    async getPhotosFromApi({ commit }) {
      commit('isLoading', true);
      const { data } = await Axios.get("https://jsonplaceholder.typicode.com/photos");
      setTimeout(() => {
        commit('savePhotosInState', data);
        commit('isLoading', false);
      }, 500);
    }
  },
})
