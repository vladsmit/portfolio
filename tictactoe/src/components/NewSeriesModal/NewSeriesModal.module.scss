.wrapper {
  width: 100%;
  height: 200px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 15px 0;
}

.inputBlock {
  width: 100%;
  display: flex;
  align-items: center;

  .textWidth {
    width: 40%;
  }

  .inputWidth {
    width: 30%;
  }

  .inputNumberWidth {
    width: 15%;
  }
}
