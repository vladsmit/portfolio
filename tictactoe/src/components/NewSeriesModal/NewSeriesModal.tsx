import { Input, InputNumber, Modal, Radio } from 'antd';
import { useState } from 'react';

import { useAppSelector, useAppDispatch } from '../../hooks';
import { TTurn } from '../../interfaces';
import {
  selectPlayers,
  updateForNewSeries,
} from '../../store/slices/ticTacToeSlice';
import styles from './NewSeriesModal.module.scss';

interface INewSeriesModalProps {
  isOpen: boolean;
  setIsOpen: (bool: boolean) => void;
}

export const NewSeriesModal = (props: INewSeriesModalProps) => {
  const { isOpen, setIsOpen } = props;

  const { cross: firstPlayer, zero: secondPlayer } =
    useAppSelector(selectPlayers);
  const dispatch = useAppDispatch();

  const [firstPlayerName, setFirstPlayerName] = useState(firstPlayer.name);
  const [secondPlayerName, setSecondPlayerName] = useState(secondPlayer.name);
  const [gameFieldSize, setGameFieldSize] = useState(3);
  const [startWith, setStartWith] = useState<TTurn>('cross');

  const startNewSeries = () => {
    dispatch(
      updateForNewSeries(
        firstPlayerName,
        secondPlayerName,
        gameFieldSize,
        startWith,
      ),
    );
    setIsOpen(false);
  };

  return (
    <Modal
      title="Новая серия"
      open={isOpen}
      onOk={startNewSeries}
      onCancel={() => setIsOpen(false)}
      okButtonProps={{ disabled: !firstPlayerName || !secondPlayerName }}
      okText="Создать"
      cancelText="Отмена"
    >
      <div className={styles.wrapper}>
        <div className={styles.inputBlock}>
          <span className={styles.textWidth}>Имя игрока 1:</span>
          <Input
            className={styles.inputWidth}
            value={firstPlayerName}
            onChange={e => setFirstPlayerName(e.target.value)}
            placeholder="Игрок 1"
          />
        </div>
        <div className={styles.inputBlock}>
          <span className={styles.textWidth}>Имя игрока 2:</span>
          <Input
            className={styles.inputWidth}
            value={secondPlayerName}
            onChange={e => setSecondPlayerName(e.target.value)}
            placeholder="Игрок 2"
          />
        </div>
        <div className={styles.inputBlock}>
          <span className={styles.textWidth}>Размер игрового поля:</span>
          <InputNumber
            className={styles.inputNumberWidth}
            value={gameFieldSize}
            onChange={value => setGameFieldSize(value ?? 3)}
            min={3}
            max={10}
          />
        </div>
        <div className={styles.inputBlock}>
          <span className={styles.textWidth}>Кто ходит первым:</span>
          <Radio.Group
            onChange={e => setStartWith(e.target.value)}
            value={startWith}
          >
            <Radio value="cross">Игрок 1</Radio>
            <Radio value="zero">Игрок 2</Radio>
          </Radio.Group>
        </div>
      </div>
    </Modal>
  );
};
