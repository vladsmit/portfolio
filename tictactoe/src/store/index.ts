import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import ticTacToeReducer from './slices/ticTacToeSlice';

export const store = configureStore({
  reducer: {
    ticTacToe: ticTacToeReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
