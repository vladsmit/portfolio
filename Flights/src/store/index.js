import { createStore, applyMiddleware, compose } from 'redux';
import { createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';

import RootReducers from './reducers';
import middlewares from './middlewares';

export const history = createBrowserHistory();

const composer = window.__REDUX_DEVTOOLS_EXTENSION__ ? compose(applyMiddleware(routerMiddleware(history), thunk, ...middlewares), window.__REDUX_DEVTOOLS_EXTENSION__(),) : compose(applyMiddleware(routerMiddleware(history), thunk, ...middlewares),);

export const store = createStore(RootReducers(history), composer,);