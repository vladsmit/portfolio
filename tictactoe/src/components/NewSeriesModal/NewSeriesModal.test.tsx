import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import { store } from '../../store';
import { NewSeriesModal } from './NewSeriesModal';

describe('NewSeriesModal component', () => {
  it('should render NewSeriesModal component', () => {
    let isOpen = true;
    const onClick = jest.fn(() => (isOpen = !isOpen));
    
    const component = render(
      <Provider store={store}>
        <NewSeriesModal isOpen={isOpen} setIsOpen={onClick} />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
