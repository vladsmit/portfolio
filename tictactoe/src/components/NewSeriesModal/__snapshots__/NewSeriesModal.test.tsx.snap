// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewSeriesModal component should render NewSeriesModal component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <style
        data-rc-order="prepend"
        rc-util-key="@ant-design-icons"
      >
        
.anticon {
  display: inline-block;
  color: inherit;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.anticon &gt; * {
  line-height: 1;
}

.anticon svg {
  display: inline-block;
}

.anticon::before {
  display: none;
}

.anticon .anticon-icon {
  display: block;
}

.anticon[tabindex] {
  cursor: pointer;
}

.anticon-spin::before,
.anticon-spin {
  display: inline-block;
  -webkit-animation: loadingCircle 1s infinite linear;
  animation: loadingCircle 1s infinite linear;
}

@-webkit-keyframes loadingCircle {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes loadingCircle {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

      </style>
      <div
        class="ant-modal-root css-dev-only-do-not-override-k7429z"
      >
        <div
          class="ant-modal-mask"
        />
        <div
          class="ant-modal-wrap"
          tabindex="-1"
        >
          <div
            aria-labelledby="test-id"
            aria-modal="true"
            class="ant-modal css-dev-only-do-not-override-k7429z ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
            role="dialog"
            style="width: 520px;"
          >
            <div
              aria-hidden="true"
              style="width: 0px; height: 0px; overflow: hidden; outline: none;"
              tabindex="0"
            />
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  Новая серия
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <div
                  class="wrapper"
                >
                  <div
                    class="inputBlock"
                  >
                    <span
                      class="textWidth"
                    >
                      Имя игрока 1:
                    </span>
                    <input
                      class="ant-input css-dev-only-do-not-override-k7429z inputWidth"
                      placeholder="Игрок 1"
                      type="text"
                      value="Игрок 1"
                    />
                  </div>
                  <div
                    class="inputBlock"
                  >
                    <span
                      class="textWidth"
                    >
                      Имя игрока 2:
                    </span>
                    <input
                      class="ant-input css-dev-only-do-not-override-k7429z inputWidth"
                      placeholder="Игрок 2"
                      type="text"
                      value="Игрок 2"
                    />
                  </div>
                  <div
                    class="inputBlock"
                  >
                    <span
                      class="textWidth"
                    >
                      Размер игрового поля:
                    </span>
                    <div
                      class="ant-input-number css-dev-only-do-not-override-k7429z inputNumberWidth"
                    >
                      <div
                        class="ant-input-number-handler-wrap"
                      >
                        <span
                          aria-disabled="false"
                          aria-label="Increase Value"
                          class="ant-input-number-handler ant-input-number-handler-up"
                          role="button"
                          unselectable="on"
                        >
                          <span
                            aria-label="up"
                            class="anticon anticon-up ant-input-number-handler-up-inner"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="up"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                        <span
                          aria-disabled="true"
                          aria-label="Decrease Value"
                          class="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
                          role="button"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-input-number-handler-down-inner"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                      <div
                        class="ant-input-number-input-wrap"
                      >
                        <input
                          aria-valuemax="10"
                          aria-valuemin="3"
                          aria-valuenow="3"
                          autocomplete="off"
                          class="ant-input-number-input"
                          role="spinbutton"
                          step="1"
                          value="3"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="inputBlock"
                  >
                    <span
                      class="textWidth"
                    >
                      Кто ходит первым:
                    </span>
                    <div
                      class="ant-radio-group ant-radio-group-outline css-dev-only-do-not-override-k7429z"
                    >
                      <label
                        class="ant-radio-wrapper ant-radio-wrapper-checked css-dev-only-do-not-override-k7429z"
                      >
                        <span
                          class="ant-radio ant-radio-checked"
                        >
                          <input
                            checked=""
                            class="ant-radio-input"
                            type="radio"
                            value="cross"
                          />
                          <span
                            class="ant-radio-inner"
                          />
                        </span>
                        <span>
                          Игрок 1
                        </span>
                      </label>
                      <label
                        class="ant-radio-wrapper css-dev-only-do-not-override-k7429z"
                      >
                        <span
                          class="ant-radio"
                        >
                          <input
                            class="ant-radio-input"
                            type="radio"
                            value="zero"
                          />
                          <span
                            class="ant-radio-inner"
                          />
                        </span>
                        <span>
                          Игрок 2
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-modal-footer"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-default"
                  type="button"
                >
                  <span>
                    Отмена
                  </span>
                </button>
                <button
                  class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-primary"
                  type="button"
                >
                  <span>
                    Создать
                  </span>
                </button>
              </div>
            </div>
            <div
              aria-hidden="true"
              style="width: 0px; height: 0px; overflow: hidden; outline: none;"
              tabindex="0"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
